server/src/modules/clubs/infra/repos/club-repo.ts

```diff
@@ -0,0 +1,7 @@
+import { Club } from '../../domain/entities/club'
+
+export abstract class ClubRepo {
+  abstract exists(name: string): Promise<boolean>
+  abstract getClubByClubId(clubId: string): Promise<Club>
```

Comment:
so yeah this signature would change to include a generic mapper

Reply:
final decision: leave it as is (no mapper) for now